apps/prairielearn/python/prairielearn.py

```diff
@@ -1440,9 +1439,10 @@ def string_to_2darray(s, allow_complex=True):
         matrix = np.zeros((number_of_rows, number_of_columns))
 
         # Parse each row and column
-        for i in range(0, number_of_rows):
-            for j in range(0, number_of_columns):
-                try:
+        i, j = 0, 0
+        try:
+            for i in range(number_of_rows):
+                for j in range(number_of_columns):
```

Comment:
Was the inversion of the `try`/`for` done because of a lint rule, or personal preference?

Reply:
That was a lint rule. "no try inside for loops" I believe.

Reply:
Huh, TIL: https://docs.astral.sh/ruff/rules/try-except-in-loop/

Not 100% convinced of the value here, but I can roll with this.